x-pack/plugins/integration_assistant/public/components/create_integration/create_integration_assistant/steps/deploy_step/use_deploy_integration.ts

```diff
@@ -46,6 +46,12 @@ export const useDeployIntegration = ({
 
     (async () => {
       try {
+        if (integrationSettings.samplesFormat == null) {
+          throw new Error(
+            'Logic error: samplesFormat is required and cannot be null or undefined when creating integration.'
```

Comment:
This would be an actual logic error (we only allow to proceed if the logs are parsed), so it should not happen. Is there a better way to express this?

Reply:
Instead make `samplesFormat` mandatory in `integrationSettings`

Reply:
How is that possible? When we just show the data stream step there are no logs provided, so no `samplesFormat`.